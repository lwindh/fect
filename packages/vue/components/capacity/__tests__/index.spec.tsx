import { mount } from '@vue/test-utils'
import Capacity from '../index'

describe('Capacity', () => {
  it('should be render as element', () => {
    const capacity = mount(Capacity)
    expect(() => capacity.unmount()).not.toThrow()
  })

  it('should render title attrs', async () => {
    const wrapper = mount({
      render() {
        return <Capacity value="40" />
      }
    })
    const el = wrapper.find('.fect-capacity')
    expect(el.attributes('title')).toBe('40%')
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.setProps({
      value: 80
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should support custom color', () => {
    const wrapper = mount({
      render() {
        return <Capacity value={'30'} color="pink" />
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
