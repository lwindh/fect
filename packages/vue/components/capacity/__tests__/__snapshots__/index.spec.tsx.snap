// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Capacity should render title attrs 1`] = `<div title="40%" class="fect-capacity"><span style="width: 40%;"></span></div>`;

exports[`Capacity should render title attrs 2`] = `<div title="80%" class="fect-capacity"><span style="width: 80%;"></span></div>`;

exports[`Capacity should support custom color 1`] = `<div title="30%" class="fect-capacity"><span style="background-color: pink; width: 30%;"></span></div>`;
